fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/ScanComparisons.java

```diff
@@ -330,6 +332,13 @@ public static Object toTupleItem(@Nullable Object item) {
             return ((Internal.EnumLite) item).getNumber();
         } else if (item instanceof FDBRecordVersion) {
             return ((FDBRecordVersion) item).toVersionstamp();
+        } else if (item instanceof DynamicMessage) {
```

Comment:
This seems brittle as a nested message will fail. We need at least a check in here to make sure all the constituent parts are primitive. Also can you make sure that iterating using the keyset of the field descriptors will indeed give you back the field descriptors in definition-order?